Feat/ssh hardening

jleivo opened the pull request:
SSH hardening configuration

- Access for jellyfin user will be limited to jellyfin1 server only
- Commands that jellyfin user can run will be limited to ffmpeg only
- Commands run by jellyfin user will be logged
    - (optional) Logs stored in separate log file

tweaks on documentation
- docs folder
- updated SETUP.MD to be copy paste friendly
- added HARDENING.md

jleivo committed:
Feat/ssh hardening

SSH hardening configuration

- Access for jellyfin user will be limited to jellyfin1 server only
- Commands that jellyfin user can run will be limited to ffmpeg only
- Commands run by jellyfin user will be logged
    - (optional) Logs stored in separate log file

tweaks on documentation
- docs folder
- updated SETUP.MD to be copy paste friendly
- added HARDENING.md

joshuaboniface reviewed (approved):
A great start! One big potential problem I see, and why I originally moved this from BASH to Python, is that FFMpeg arguments will regularly be mangled by Bash, especially quotes on filenames. I would suggest rewriting the limitation wrapper in Python for that reason. But otherwise since this is all documentation/examples, happy to get it in!

jleivo committed:
feat: Added python version of the wrapper

- GPT-OSS:120b did the converson of the script from bash to python3
- Updated rsyslog definition to work with either bash or python -file
- updated SSH config to use the python version
